HGCalAnalysis/plugins/HGCalAnalysis.cc

```diff
@@ -270,7 +305,10 @@ HGCalAnalysis::HGCalAnalysis(const edm::ParameterSet& iConfig) :
 	propagationPtThreshold(iConfig.getUntrackedParameter<double>("propagationPtThreshold",3.0)),
 	detector(iConfig.getParameter<std::string >("detector")),
 	rawRecHits(iConfig.getParameter<bool>("rawRecHits")),
-	particleFilter(iConfig.getParameter<edm::ParameterSet>("TestParticleFilter"))
+	particleFilter(iConfig.getParameter<edm::ParameterSet>("TestParticleFilter")),
+	dEdXWeights(iConfig.getParameter<std::vector<double> >("dEdXWeights")),
+	invThicknessCorrection({1./1.132,1./1.092,1./1.084}),
```

Comment:
I have prepared the modification in HGCalRecHit_cfi so that these parameters can be extracted. It is in 
https://github.com/beaudett/cmssw/tree/topic_extract_parameters_3
That would need to go first in CMSSW before we can use it in reco-ntuples.